pytorch_lightning/loops/batch/training_batch_loop.py

```diff
@@ -211,37 +213,55 @@ def _run_optimization(
         self._run_optimization_end(opt_idx)
         return result
 
-    def _training_step_and_backward_closure(
-        self,
-        split_batch: Any,
-        batch_idx: int,
-        opt_idx: int,
-        optimizer: Optimizer,
-        hiddens: Tensor,
-        return_result: AttributeDict,
-    ) -> Optional[Tensor]:
-        """Closure for training step and backward
+    def _make_closure(self, split_batch, batch_idx, opt_idx, optimizer, hiddens):
+        step_fn = self._make_step_fn(split_batch, batch_idx, opt_idx, hiddens)
+        backward_fn = self._make_backward_fn(batch_idx, optimizer, opt_idx)
+        zero_grad_fn = self._make_zero_grad_fn(batch_idx, opt_idx, optimizer)
```

Comment:
Should the definition of these functions become part of `LightningClosure`

If the `LightningClosure` becomes the smallest unit of optimization, it means that it can be reused across loops of different semantic meanings.

In that case, It would be nice to be able to subclass the closure implementation without any changes to the loops

Reply:
I think one benefit of having these outside of `LightningClosure` is we can creating different kind of closures (for handing different `training_step` signatures and optimization flows) without requiring `LightningClosure` to be aware of all of them.